src/python/pants/core/goals/partitions.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
should be in `util_rules`. `goals/` is for things that show up in `./pants help goals`